independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/ResteasyReactiveRequestContext.java

```diff
@@ -161,9 +171,17 @@ public ProvidersImpl getProviders() {
      * @param target The resource target
      */
     public void restart(RuntimeResource target) {
+        restart(target, false);
+    }
+
+    public void restart(RuntimeResource target, boolean setLocatorTarget) {
         this.handlers = target.getHandlerChain();
         position = 0;
         parameters = new Object[target.getParameterTypes().length];
+        if (setLocatorTarget) {
```

Comment:
Can a locator method return a sub-resource which returns a locator method? As in, can we have two levels of locators? If yes, then we need more than an extra field, we need a stack of them.

Reply:
That's a very good question, but the spec doesn't seem to say anything about that nor do I see any test in the TCK.

I would say let's not complicate it unless we actually see something like that in the wild

Reply:
+1